src/Microsoft.TemplateEngine.Cli/Commands/InstantiateCommand.TabCompletion.cs

```diff
@@ -11,27 +12,36 @@ namespace Microsoft.TemplateEngine.Cli.Commands
 {
     internal partial class InstantiateCommand : BaseCommand<InstantiateCommandArgs>
     {
-        internal IEnumerable<string> GetSuggestions(
+        internal IEnumerable<CompletionItem> GetCompletions(
             InstantiateCommandArgs args,
             IEnumerable<TemplateGroup> templateGroups,
             IEngineEnvironmentSettings environmentSettings,
             TemplatePackageManager templatePackageManager,
-            string? textToMatch)
+            TextCompletionContext context)
         {
             if (string.IsNullOrWhiteSpace(args.ShortName))
             {
-                return templateGroups.SelectMany(g => g.ShortNames).Distinct().OrderBy(s => s, StringComparer.OrdinalIgnoreCase);
+                return templateGroups
+                    .SelectMany(g => g.ShortNames, (g, shortName) => new CompletionItem(shortName, documentation: g.Description))
+                    .Distinct()
```

Comment:
`CompletionItem` is a class. I suspect calling `Distinct` here won't do much good for us since each instance will be unique even though they contain the same `shortName`. Maybe use `DistinctBy(c => c.Label)`?

Reply:
Same goes for other usages of `Distinct` below.

Reply:
added here, and in other cases

Reply:
as per @jonsequitur comment below, I removed custom comparer. Instead added test which will detect change in behavior in case behavior in System.CommandLine will change.

Reply:
I am a bit confused on which of Jon's comments are you talking about here. I thought you meant the one about "keeping the duplicates in the suggestions", but now I think you mean about an existing custom comparer in the library.

If we don't want duplicates, than this code is fine. If we want duplicates, we can remove `Distinct` calls.